internal/clients/service.go

```diff
@@ -22,28 +32,61 @@ func NewTemporalService(configData []byte) (*TemporalServiceImpl, error) {
 	var conf = TemporalServiceConfig{}
 	err := json.Unmarshal(configData, &conf)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to unmarshal config data: %w", err)
 	}
 
 	logger := slog.New(slog.NewJSONHandler(os.Stdout, &slog.HandlerOptions{
 		AddSource: true,
 		Level:     slog.LevelDebug,
 	}))
 
+	logger.Debug("Starting NewTemporalService", slog.String("hostPort", conf.HostPort), slog.Bool("useTLS", conf.UseTLS))
+
+	var dialOptions []grpc.DialOption
+	if conf.UseTLS && conf.CACert != "" && conf.CertFile != "" && conf.KeyFile != "" {
+		logger.Debug("Loading client certificate from strings")
+		cert, err := tls.X509KeyPair([]byte(conf.CertFile), []byte(conf.KeyFile))
+		if err != nil {
+			return nil, fmt.Errorf("failed to load client certificate: %w", err)
```

Comment:
can you plz use `errors.Wrap(err, "some detail err msg")`  (also in the other occurences)
Because it's the default for crossplane providers (see [here](https://github.com/denniskniep/provider-temporal/blob/c7b721c1e73556d9f268123eb0967665e3bf75c3/internal/controller/searchattribute/searchattribute.go#L28C1-L29C1) )

We could think about changing this, because this lib's repo is archived. But then we should change also all other occurrences so that we use the same approach in the entire project.

Reply:
@mihaelabalas84 can you also address this comment ?

Else it looks good to me

Reply:
done @denniskniep